rotkehlchen/chain/ethereum/modules/juicebox/decoder.py

```diff
@@ -0,0 +1,127 @@
+    PAY_ABI,
+    PAY_SIGNATURE,
+    TERMINAL_3_1_2,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JuiceboxDecoder(DecoderInterface):
+
+    def _query_project_name(self, project_id: int) -> dict[str, Any] | None:
+        """Query metadata for project id in ipfs"""
+        try:
+            ipfs_hash = self.evm_inquirer.contracts.contract(JUICEBOX_PROJECTS).call(
+                node_inquirer=self.evm_inquirer,
+                method_name='metadataContentOf',
+                arguments=[project_id, 0],  # 0 is the ipfs hash of the data
+            )
+        except RemoteError:
+            log.error(f'Failed to query ipfs hash for juicebox project {project_id}')
+            return None
+
+        metadata_url = JUICEBOX_IPFS_GATEWAY + ipfs_hash
+        try:
+            log.debug(f'Querying juicebox metadata at {metadata_url}')
+            metadata: dict[str, Any] = requests.get(
+                url=metadata_url,
+                timeout=CachedSettings().get_timeout_tuple(),
+            ).json()
+        except requests.RequestException as e:
+            log.error(f'Failed to query juicebox project {project_id} metadata due to {e}')
+            return None
+
+        return metadata.get('name')
+
+    def _decode_pay(self, context: DecoderContext) -> DecodingOutput:
+        """Decode pay with rewards in juicebox"""
+        if context.tx_log.topics[0] != PAY_SIGNATURE:
+            return DEFAULT_DECODING_OUTPUT
+
+        try:
+            topic_data, decoded_data = decode_event_data_abi_str(context.tx_log, PAY_ABI)
+        except DeserializationError as e:
+            log.error(
+                f'Failed to deserialize Juicebox event at '
+                f'{context.transaction.tx_hash.hex()} due to {e}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=decoded_data[2],
+            token_decimals=18,  # it is always ETH
+        )
+        project_id = topic_data[2]
+        project_name = self._query_project_name(project_id=project_id)
+        is_donation = project_id == LEGAL_DEFENSE_TORNADO_PROJECT_ID
```

Comment:
As I said the first time we talked can you see any of the other projects in the list to see if anything else is also a donation campaign and classify them as such?